geopandas/base.py

```diff
@@ -88,13 +88,23 @@ def _delegate_property(op, this):
         return Series(data, index=this.index)
 
 
-def _delegate_geo_method(op, this, *args, **kwargs):
+def _delegate_geo_method(op, this, **kwargs):
     # type: (str, GeoSeries) -> GeoSeries
     """Unary operation that returns a GeoSeries"""
     from .geoseries import GeoSeries
 
+    for key, val in kwargs.items():
+        if isinstance(val, pd.Series) and not val.index.equals(this.index):
+            raise ValueError(
+                f"Index of the Series passed as '{key}' does not match index of the "
+                "GeoSeries. If you want both Series to be aligned, align them before "
+                f"passing them to this method as `this, {key} = this.align({key})`. If "
+                f"you want to ignore the index pass the underlying array as '{key}' "
+                f"using `{key}.values`."
+            )
```

Comment:
So `this` is not necessarily a GeoSeries (contrary to the type comment above), it's just a subtype of GeopandasBase, so referring to GeoSeries in the body is potentially misleading if `gdf.concave_hull(ratio=...)` is called. I'd suggest we do something like this to try and be a bit more friendly:
```suggestion
    from .geodataframe import GeoDataFrame

    if isinstance(this, GeoSeries):
        klass, var_name = "GeoSeries", "g"
    elif isinstance(this, GeoDataFrame):
        klass, var_name = "GeoDataFrame", "gdf"
    else:
        klass, var_name = this.__class__.__name__, "this"

    for key, val in kwargs.items():
        if isinstance(val, pd.Series) and not val.index.equals(this.index):
            raise ValueError(
                f"Index of the Series passed as '{key}' does not match index of the "
                f"{klass}. If you want both Series to be aligned, align them before "
                f"passing them to this method as "
                f"`{var_name}, {key} = {var_name}.align({key})`. If "
                f"you want to ignore the index, pass the underlying array as '{key}' "
                f"using `{key}.values`."
            )
```
(we could also not refer to GeoSeries/GeoDataFrame at all and just refer to this, but I think there's value in trying to have more descriptive language for the common case.)